src/lib.rs

```diff
@@ -137,7 +193,7 @@ where
                     match &mut node[i] {
                         None => (),
                         Some(annotated) => {
-                            match annotated.val_mut()?._push(t)? {
+                            match annotated.compound_mut()?._push(t)? {
```

Comment:
Just a question, why the naming with the underscore for `_push` and `_pop`?

Reply:
It's to distinguish it from the user-facing pub function. could be called inner_pop etc but this is good enough imo.